src/api/spec/README.md

```diff
@@ -45,6 +45,85 @@ We are using the standard [RSpec generators](https://relishapp.com/rspec/rspec-r
+Be aware of that factories, other than fixtures, run through ActiveRecord validations.
+All OBS factories reside in `spec/factories`.
+
+#### has_many associations
+For creating has_many associations we prefer ```create_list```:
+
+```
+project.packages = create_list(:package, 2)
+```
+
+Please also have a look at the [factory girl documentation](https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#associations)
+
+#### Use a sequence for unique values
+It's necessary to use a [sequence](https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#sequences) for attributes which have to be unique like project.title or user.login.
+
+```
+sequence(:login) { |n| "user_#{n}" }
+```
+
+Please keep in mind that you have to overwrite these attributes if they are part of the URI and you use it in combination with VCR.
+Otherwise your tests will fail as VCR matches the cassette by the URI.
+
+```
+let!(:user) { create(:confirmed_user, login: "proxy_user") }
+```
+
+By passing ```login: "proxy_user")``` to the create statement, the username is now always proxy_user and not random (e.g. user_42).
+
+#### Factories should be the bare minimum
+Different to fixtures, factory girl runs through your ActiveRecord validations.
+That said, only add the bare minimum to your factory which is required to be valid.
+You can use an inherited factory to add or override attributes.
+
+```
+  factory :user do
+    email { Faker::Internet.email }
+    realname { Faker::Name.name }
+    sequence(:login) { |n| "user_#{n}" }
+    password 'buildservice'
+
+    factory :confirmed_user do
+      state 2
+    end
+```
+
+See this [blog article](https://robots.thoughtbot.com/factories-should-be-the-bare-minimum) for a detailed explanation.
+
+#### When Transient Attributes make sense
+Use [transient attributes](https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#transient-attributes) to DRY your factories.
+
+```
+  factory :project_with_package do
+    transient do
+      sequence(:package_name) { |n| "package_#{n}" }
+      create_patchinfo false
+      maintainer nil
+    end
+
+    after(:create) do |project, evaluator|
+      project.packages << Package.where(project_id: project.id, name: evaluator.package_name) || create(:package, project_id: project.id, name: evaluator.package_name)
+    end
+  end
+```
+
+Without the transient attribute package_name it would be necessary to explicit create a package with a different name.
+Now you can just do:
+
+```
+create(:project_with_package, package_name: 'foobar')
+```
+
+#### Generating fake data
+We use the [faker gem](https://github.com/stympy/faker) to generate more realistic test data.
+However, we don't use this in cases where we use the data to identify objects (like user.login or project.title), to simplify debugging.
+In that case, please use a simple sequence.
```

Comment:
Currently we use it on several places. And if we don't mention it I expect that some people start to write eg ``lorem ipsum`` instead of using faker!

I thought we just agreed last week that we don't want to use it for identifier anymore?